### PR TITLE
Updates db indexes to improve performance

### PR DESCRIPTION
**Pull request by mgdelacroix**
#### Summary
This changes update and add some focalboard indexes to improve data access speed. The rationale for the changes is as follows:

- `focalboard_blocks (channel_id, id)` was not in use at all after the permission changes, so it doesn't make sense to keep as the PK of the table.
- `focalboard_blocks (id)` makes sense now as a PK, and improves:
  - `getBlocksByIDs` (MySQL 0.215 => 0.001 | PostgreSQL 87.063 ms => 0.082 ms)
- `idx_blocks_board_id_parent_id` improves:
  - `getWithParentAndType` (MySQL 0.197 => 0.001 | PostgreSQL 81.356 ms => 0.068 ms)
  - `getBlocksWithParent` (MySQL 0.195 => 0.001 | PostgreSQL 93.420 ms => 0.055 ms)
  - `getSubTree2` (MySQL 0.162 => 0.001 | PostgreSQL 88.920 ms => 0.092 ms)
  - `getBlocksWithType` (MySQL 0.215 => 0.003 | PostgreSQL 84.725 ms => 0.055 ms)
  - `getBlocksForBoard` (MySQL 0.194 => 0.001 | PostgreSQL 85.891 ms => 0.058 ms)
- `idx_subscriptions_subscriber_id` covers subscription fetching for a given user, which is run once per board load on an otherwise full scan of the table:
  - `getSubscriptions` (MySQL 0.023 => 0.001 | PostgreSQL 14,559 ms => 1.838 ms)

These tests have been run in plugin mode, on the following datasets:
- MySQL
  - Boards: 74827
  - Blocks: 299446
  - Users: 301
  - Subscriptions: 74790

- PostgreSQL
  - Boards: 145411
  - Blocks: 582028
  - Users: 982
  - Subscriptions: 145404

And the time the indexes took to apply was of 13.2615s for MySQL and 3.3575s for PostgreSQL.

#### Card Link
https://community.mattermost.com/boards/team/qghzt68dq7bopgqamcnziq69ao/b4ctmm917q3rzdfi451qtuxjzcw/612a901f-5c6d-4dde-8e35-3c16b6024a67/c94wm8nfiftfbbng3ni6dpb971a

### CODE REVIEW

File: server/services/store/storetests/blocks.go [hunk 1]
```diff
@@ -966,12 +966,12 @@ func testGetBlockMetadata(t *testing.T, store store.Store) {
 	})
 
 	t.Run("get block history after updateAt", func(t *testing.T) {
-		rBlocks, err2 := store.GetBlocksWithType(boardID, "test")
```
Comment (mgdelacroix) [inline server/services/store/storetests/blocks.go:969]: These tests were relying on the order of the full-scan to work properly, which now changes with the index. They've been updated to directly access the specific block needed

**Review by sbishel — APPROVED** [review]
LGTM

**Comment by wiggin77** [comment]
> getSubscriptions (MySQL 0.023 => 0.001 | PostgreSQL 14,559 ms => 1.838 ms)

Did we get any clarity on this number.  Was it really 14.6 seconds?

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:  Great improvements in the benchmarks.  Looking forward to seeing this on community.

**Comment by mgdelacroix** [comment]
> > getSubscriptions (MySQL 0.023 => 0.001 | PostgreSQL 14,559 ms => 1.838 ms)
> 
> Did we get any clarity on this number. Was it really 14.6 seconds?

I think that's the Spanish locale jumping in and messing with the output of `\timings`, so I ended up using `EXPLAIN ANALYZE` to both have the output in the right format and remove the (almost negligible) network time of the query. Definitely 14ms

**Pull request merged**